source/core/slang-defines.h

```diff
@@ -0,0 +1,459 @@
+#ifndef SLANG_DEFINES_H
```

Comment:
This whole file seems good, and we probably need to merge some of this with the existing definitions for `SLANG_ASSERT` and friends.

I'd caution against adding too much "infrastructure" code without any end-user-visible functional improvements to pay for it, but at the same time you are cranking this out fast enough that it isn't a huge time-sink.